backend/tests/integration/records/views/test_views.py

```diff
@@ -148,3 +151,117 @@ def test_returns_301_with_proper_location_when_record_redirected_in_chain(inspir
 
     assert response.status_code == 301
     assert response.location.split("/")[-1] == new_cn
+
+
+def test_literature_workflows_record_source(inspire_app):
```

Comment:
just a suggestion, it would be more clear if we have separated tests for each case